### PR TITLE
ensure datachannel ready before publish data

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

**Comment by cnderrauber** [comment]
also make room callbacks to be async. fix #49

File: engine.go [hunk 1]
```diff
@@ -243,12 +243,12 @@ func (e *RTCEngine) waitUntilConnected() error {
 	}
 }
 
-func (e *RTCEngine) ensurePublisherConnected() error {
+func (e *RTCEngine) ensurePublisherConnected(dataReady bool) error {
```
Comment (boks1971) [inline engine.go:246]: Maybe rename this to `checkDataReady`. Just `dataReady` feels like it is a signal about readiness of data channel. Or maybe `ignoreDataReady`.

**Review by boks1971 — APPROVED** [review]

File: engine.go (hunk 1 above)
Comment (cnderrauber) [reply engine.go]: done

**Comment by cnderrauber** [comment]
I'm not sure, both pion and lk-server use OpsQueue in internal logic, will observe this part, can change to use OpsQueue if we get performance/timing problem with go func()

**Pull request merged**